src/ppx/scene/scene_material.cpp

```diff
@@ -0,0 +1,219 @@
+// Copyright 2023 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "ppx/scene/scene_material.h"
+#include "ppx/grfx/grfx_device.h"
+
+namespace ppx {
+namespace scene {
+
+// -------------------------------------------------------------------------------------------------
+// Image
+// -------------------------------------------------------------------------------------------------
+Image::Image(
+    grfx::Image*            pImage,
+    grfx::SampledImageView* pImageView)
+    : mImage(pImage),
+      mImageView(pImageView)
+{
+}
+
+Image::~Image()
+{
+    if (mImageView) {
+        auto pDevice = mImageView->GetDevice();
+        pDevice->DestroySampledImageView(mImageView);
+        mImageView.Reset();
```

Comment:
this is not needed right? Since the pointer destructor will be called at the end.

Reply:
Left over game programmer paranoia. Removed.